src/InPluginUpsells/SaleBanners.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Give\InPluginUpsells;
+
+/**
+ * @unreleased
+ */
+class SaleBanners
+{
+	/**
+	 * @var string
+	 */
+	private $optionName = 'hidden-sale-banners';
+
+	/**
+	 * @var array
+	 */
+	private $hiddenBanners;
+
+	public function __construct()
+	{
+		$this->hiddenBanners = give_get_option($this->optionName, []);
+	}
+
+	/**
+	 * Get banners definitions
+	 *
+	 * @note id must be unique for each definition
+	 *
+	 * @return array[]
+	 */
+	public function getBanners()
+	{
+		return [
+			[
+				'id'              => 'bfgt2021',
+				'accessibleLabel' => __('Black Friday/Giving Tuesday Sale', 'give'),
+				'leadText'        => __('Save 40% on all plans for a limited time.', 'give'),
+				'contentText'     => __('Black Friday through Giving Tuesday.', 'give'),
+				'actionText'      => __('Shop Now', 'give'),
+				'actionURL'       => 'https://givewp.com/sale',
+				'startDate'       => strtotime('2021-11-14 00:00'),
+				'endDate'         => strtotime('2021-11-17 24:00'),
+			],
+		];
+	}
+
+	/**
+	 * Get the banners that should be displayed.
+	 *
+	 * @return array[]
+	 */
+	public function getVisibleBanners()
+	{
+		$currentDateTime = current_datetime()->getTimestamp();
+		$currentUserId = get_current_user_id();
+
+		return array_filter(
+			$this->getBanners(),
+			function ($banner) use ($currentDateTime, $currentUserId) {
+				$isHidden = in_array($banner['id'] . "_{$currentUserId}", $this->hiddenBanners);
```

Comment:
Underscore between `$banner['id']` and `$currentUserId` should be removed. Otherwise, the hide banner action won't work.

Reply:
Resolved in 2f377f3